proxy/tests/telemetry.rs

```diff
@@ -98,6 +109,176 @@ fn http1_inbound_sends_telemetry() {
+
+    info!("client.get(/hey)");
+    assert_eq!(client.get("/hey"), "hello");
+
+    info!("client.get(/hi)");
+    assert_eq!(client.get("/hi"), "good morning");
+    assert_eq!(client.get("/hi"), "good morning");
+
+    info!("awaiting report");
+    let report = reports.wait().next().unwrap().unwrap();
+    // proxy inbound
+    assert_eq!(report.proxy, 0);
+    // process
+    assert_eq!(report.process.as_ref().unwrap().node, "");
+    assert_eq!(report.process.as_ref().unwrap().scheduled_instance, "");
+    assert_eq!(report.process.as_ref().unwrap().scheduled_namespace, "");
+
+    // requests -----------------------
+    assert_eq!(report.requests.len(), 2);
+
+    // -- first request -----------------
+    let req = &report.requests[0];
+    assert_eq!(req.ctx.as_ref().unwrap().authority, "test.conduit.local");
+    assert_eq!(req.ctx.as_ref().unwrap().path, "/hey");
+    assert_eq!(req.count, 1);
+    assert_eq!(req.responses.len(), 1);
+    // ---- response --------------------
+    let res = &req.responses[0];
+    assert_eq!(res.ctx.as_ref().unwrap().http_status_code, 200);
+    // response latencies should always have a length equal to the number
+    // of latency buckets in the latency histogram.
+    assert_eq!(
+        res.response_latency_counts.len(),
+        report.histogram_bucket_bounds_tenth_ms.len()
+    );
+    assert_eq!(res.ends.len(), 1);
+
+    // ------ ends ----------------------
+    let ends = &res.ends[0];
+    assert_eq!(ends.streams.len(), 1);
+    // -------- streams -----------------
+    let stream = &ends.streams[0];
+    assert_eq!(stream.bytes_sent, 5);
+    assert_eq!(stream.frames_sent, 1);
+
+    // -- second request ----------------
+    let req = &report.requests[1];
+    assert_eq!(req.ctx.as_ref().unwrap().authority, "test.conduit.local");
+    assert_eq!(req.ctx.as_ref().unwrap().path, "/hi");
+    // repeated twice
+    assert_eq!(req.count, 2);
+    assert_eq!(req.responses.len(), 1);
+    // ---- response  -------------------
+    let res = &req.responses[0];
+    assert_eq!(res.ctx.as_ref().unwrap().http_status_code, 200);
+    // response latencies should always have a length equal to the number
+    // of latency buckets in the latency histogram.
+    assert_eq!(
+        res.response_latency_counts.len(),
+        report.histogram_bucket_bounds_tenth_ms.len()
+    );
+    assert_eq!(res.ends.len(), 1);
+
+    // ------ ends ----------------------
+    let ends = &res.ends[0];
+    assert_eq!(ends.streams.len(), 2);
+
+    // -------- streams -----------------
+    let stream = &ends.streams[0];
+    assert_eq!(stream.bytes_sent, 12);
+    assert_eq!(stream.frames_sent, 1);
+
+}
+
+#[test]
+fn records_latency_statistics() {
+    let _ = env_logger::init();
+
+    info!("running test server");
+    let srv = server::new()
+        .route_with_latency("/hey", "hello", Duration::from_millis(500))
+        .route_with_latency("/hi", "good morning", Duration::from_millis(40))
+        .run();
```

Comment:
we probably should disable this test by default, since it relies on timing behaviors which are notoriously flakey in CI, etc.   The 40ms sleep which actually end up being 80ms and the test will break.

Reply:
done 1db70636085c8fa910b40236070ac806b9ea3b5d